server/app.py

```diff
@@ -61,20 +63,40 @@ def quote(id):
     quote = Quote.query.get(id)
     return jsonify(quote.serialize)
 
-@app.route('/backtester/run', methods=['POST'])
-def run_backtester():
-    post_data = request.get_json()
-    params = "".join(str(post_data['params']).split())
+def execute_backtest():
+    global executing, backtest_queue
+
+    executing = True
+    args = backtest_queue.get()
+
     exe = os.path.dirname(os.path.abspath(__file__)) + '/../backtester/backtester/Release/backtester.exe'
-    proc = subprocess.Popen([exe, '--algoName', post_data['algorithm'], '--params', params, '--tickers'] +
-                            post_data['tickers'], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    proc = subprocess.Popen([exe, '--algoName', args['algorithm'], '--params', args['params'], '--tickers'] +
+                            args['tickers'], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
     while 1:
         line = proc.stdout.readline().rstrip('\r\n')
         if line == 'BACKTESTER DONE':
             break
         simulation_results = json.loads(line)
         # TODO: convert simulation_results into DB models and upload them to the DB
-    return jsonify({'BACKTESTER RUN SUCCESSFUL': "TRUE"})
+
+    if backtest_queue.qsize() > 0:
+        # starts the next enqueued backtest
+        execute_backtest()
+
+
+@app.route('/backtester/run', methods=['POST'])
+def run_backtester():
+    global executing, backtest_queue
+
+    post_data = request.get_json()
+    post_data['params'] = "".join(str(post_data['params']).split())
+
+    backtest_queue.put(post_data)
+
+    if executing is False:
+        socketio.start_background_task(target=execute_backtest)
```

Comment:
This is to prevent the simulation execution from blocking the standard HTTP request/response. Not sure how we can handle that without spawning a new thread. Any suggestions?